presto-orc/src/main/java/com/facebook/presto/orc/writer/MapFlatColumnWriter.java

```diff
@@ -228,15 +232,15 @@ public Map<Integer, ColumnStatistics> finishRowGroup()
         }
         Arrays.fill(valueWritersLastRow, 0);
 
-        ColumnStatistics mapStats = new ColumnStatistics((long) nonNullRowGroupValueCount, null);
-        rowGroupColumnStatistics.add(mapStats);
+        Map<Integer, ColumnStatistics> columnStatistics = getColumnStatisticsFromValueWriters(ColumnWriter::finishRowGroup, nonNullRowGroupValueCount);
```

Comment:
Creation of stats for the map node moved to the getColumnStatisticsFromValueWriters.